wsmaster/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java

```diff
@@ -199,6 +207,41 @@ public void shouldRemoveMachineFromRegistryIfInstanceDestroyingFailsOnDestroy()
         }
     }
 
+    @Test(expectedExceptions = IOException.class, expectedExceptionsMessageRegExp = "Stream closed")
+    public void shouldCloseProcessLoggerOnExec() throws Exception {
+        Command command = mock(Command.class);
+        when(command.getCommandLine()).thenReturn("CommandLine");
+        when(command.getName()).thenReturn("CommandName");
+        when(command.getType()).thenReturn("CommandType");
+        when(machineRegistry.getInstance(MACHINE_ID)).thenReturn(instance);
+        InstanceProcess instanceProcess = mock(InstanceProcess.class);
+        when(instance.createProcess(command, "outputChannel")).thenReturn(instanceProcess);
+        when(instanceProcess.getPid()).thenReturn(111);
+        ArgumentCaptor<LineConsumer> argumentCaptor = ArgumentCaptor.forClass(LineConsumer.class);
+        //Create log file
+        File logDir = new File(machineLogsDir + "/" + MACHINE_ID);
+        logDir.mkdirs();
+        new File(logDir, "111").createNewFile();
+
+        manager.exec(MACHINE_ID, command, "outputChannel");
+        //Wait until executor will complete the task
+        for (int i = 0; ((ThreadPoolExecutor)manager.executor).getCompletedTaskCount() == 0 && i < 10; i++) {
+            Thread.sleep(300);
+        }
+
+        verify(instanceProcess).start(argumentCaptor.capture());
+        LineConsumer lineConsumer = argumentCaptor.getValue();
+        lineConsumer.writeLine("line");
+        //Should throw exception because lineConsumer is closed
+        ((LineConsumer[])getPrivateField(lineConsumer, "lineConsumers"))[0].writeLine("line");
```

Comment:
I think it is better to make method **getProcessLogger** package private (don't forget visiblefortest annotation) and mock it. With current approach you expose executor, rely on io exception when line consumer is closed - it is very tricky.